### DISCUSSION TITLE
PCW8256 FlashFloppy Gotek boot

### DISCUSSION THREAD

**Discussion by kemalhy** [keirf/flashfloppy#899] [Q&A]
Hi,
I have a PCW8256 and I can boot from the physical drive. But I couldn't boot from Gotek. and can't read as drive B either, I modified drive S1 selection on PCW drive board just for testing. Normally I am using the drive as A: (S0)

FF version is 3.42 and config files attached.
cable connection gotek to PCW as in the picture. jumper setting MO,S0 and JB. 
I also have a greaseweazel 4.1. Used jumper wires and also tried with a csutom cable gotek to greaseweazel but no chance to read the device. I am getting following message
% gw read --device=/dev/cu.usbmodem14201 --drive=0 --tracks=c=0-39:h=0:step=1 --revs=3 image.scp
Reading c=0-39:h=0 revs=3
Command Failed: Seek: Track 0 not found

AMSTRAD	SHUGART GOTEK
1..25       GND	       1..33
2	       IND	         8
4	       DS0	        10
6	       DS1	        12
8	       MTR	        16
10	       DIR	        18
12	       STEP	        20
14	      WDATA	22
16	      WGATE	24
18	      TRK0	        26
20	      WRPROT	28
22	      RDATA	30
24	      SIDE	        32
26	      RDY	        34

[files.zip](https://github.com/keirf/flashfloppy/files/15339847/files.zip)

**Comment by keirf** [comment]
Probably getting Greaseweazle and Gotek talking would be the best first step. You shouldn't need a custom cable for this. Both have normal 34-pin headers, so you can just use a straight ribbon cable, or a PC ribbon cable (with twist on pins 10-16).

So how is the Greaseweazle-to-Gotek currently set up, exactly?

**Reply by kemalhy** [reply]
it is in pictureI attached, its not a full 34-way cable, because I only have 26-way ribbon, but I have enough jumper wires, if required I can connect full signal pins and ground. If I can set gotek with jumper S1, I think I don't need to twist the signal pins on 10-16. I tried these combinations but not with full signal wires.

**Reply by keirf** [reply]
Oh I see. That looks plausible, though maybe I don't trust the custom ribbon. Is the Gotek powered on okay (eg. display is lit)?

**Reply by kemalhy** [reply]
gotek is working fine, OLED, menu buttons and also added a rotary switch. S0 and S1 jumpers tested with greaseweazel connected. When I use 'gw read --device=/dev/cu.usbmodem14201 --drive=<0/1>' it is litting gotek LED if relevant jumper is set.

I reset the gotek config factory default to test if FF.CFG read correctyl, which is OK, my rotary switch requires "half" the default is full I guess. So gotek taking the FF.CFG. 

but I am not sure if I correctly set the IMC.CFG for Amstrad. 
I have only below settings in IMG.CFG
# Amstrad PCW 3 inch disk.
cyls = 40
heads = 1
step = 1
secs = 9
bps = 512
rpm = 300

**Reply by keirf** [reply]
It is surprising then that you get "Track 0 not found" from Greaseweazle. /TRK0 should be asserted from power on, since the Gotek powers on at Track 0.

What sort of disk images are you using? I think PCW often uses EDSK image files?

**Reply by keirf** [reply]
Oh, also, you don't need a jumper at JC. If you've specified `interface=shugart` then actually JC has no effect at all, regardless of a jumper being installed or not.

**Reply by kemalhy** [reply]
when I analyze the disk image 'CPM.dsk' in HxCFloppyEmulator app, it show the disk with following attributes:
side 0, 40 Tracks
360 Sectors, 0 bad
184320 Bytes
ISO MFM
and sector allocation such as: 1 6 2 7 3 8 4 9 5

Track RPM : 300 RPM
Bitrate : 250000 bit/s
Track format :
ISOIBM_MFM_ENCODING
Track len : 100064 cells
Number of side : 1
Interface mode:
CPC_DD_FLOPPYMODE
Amstrad CPC Interface

I tried with and without the JB jumper.

**Reply by keirf** [reply]
FlashFloppy will work with your CPM.dsk directly, and that image is self describing so no IMG.CFG is needed at all. If you are working with EDSK image files exclusively, you can just delete your IMG.CFG file entirely.

You mean jumper JC not JB? JB is an output for a speaker, you shouldn't put a jumper there at all.

I'm not sure why it's not working, especially with Greaseweazle. I'd probably suggest to get a full ribbon cable and try again there. As if it doesn't work, perhaps something is wrong with the Gotek?

**Comment by kemalhy** [comment]
my bad, I meant JC, my guess also something wrong with gotek hardware.
thanks for the detailed answers.

**Comment by Gary-Clark** [comment]
This is the FF.CFG file I used for a 9512 which is very similar, no IMG.CFG required.

interface = jc
host = unspecified
pin02 = auto
pin34 = auto
write-protect = no
side-select-glitch-filter = 0
track-change = instant
index-suppression = yes
head-settle-ms = 12

ejected-on-startup = no
image-on-startup = last
display-probe-ms = 3000
autoselect-file-secs = 1
autoselect-folder-secs = 2
nav-mode = default
nav-loop = yes
twobutton-action = zero
rotary = none
display-type = oled-128x32
oled-font = 6x13
oled-contrast = 143
display-off-secs = 255
display-on-activity = yes
display-scroll-rate = 200
display-scroll-pause = 2000
nav-scroll-rate = 80
nav-scroll-pause = 300
step-volume = 10
da-report-version = ""
extend-image = yes

**Reply by kemalhy** [reply]
Hi Gary,
thank you for the update, I tried with your FF.CFG, but the result still same. I am getting 'Track 0 not found message'
I believe I have a harware issue in gotek, I will update when new gotek arrive.

**Reply by kemalhy** [reply]
with the new Gotek it worked at the first try. Thank you for your help again.
